heads/src/main/java/com/dumbdogdiner/stickyapi/bukkit/item/SkullBuilder.java

```diff
@@ -0,0 +1,160 @@
+        this.category = category.toUpperCase();
+        return this;
+    }
+
+    /**
+     * Sets the specific head (from <code>textures.yml</code>)
+     *
+     * @param head The head to set
+     * @throws IllegalArgumentException if the head does not exist
+     * @see TextureHelper#getTexture(String, String)
+     */
+    public @NotNull SkullBuilder head(@NotNull String head) {
+        head = head.toUpperCase();
+        Preconditions.checkNotNull(TextureHelper.getTexture(category, head),
+                "The specified head %s is not a valid head", head);
+        this.head = head;
+        this.texture = TextureHelper.getTexture(category, head);
+        return this;
+    }
+
+
+    /**
+     * Sets the quantity of items in the final item stack
+     *
+     * @param i the requested quantity
+     * @throws IllegalArgumentException if the stack size is invalid
+     */
+    public @NotNull SkullBuilder quantity(int i) {
+        Preconditions.checkArgument(i >= 0 && i <= 64,
+                "Invalid stack size of %s specified  (must be between 0 and 64, inclusive)", i);
+        this.quantity = i;
+        return this;
+    }
+
+
+    /**
+     * @param textureURL A {@link URL} where a valid PNG minecraft texture can be found
+     * @return The current {@link SkullBuilder} instance
+     * @throws IllegalArgumentException if the URL is invalid
+     */
+    public @NotNull SkullBuilder texture(@NotNull URL textureURL) {
+        TextureValidator.validateTextureUrl(textureURL.toExternalForm());
+
+        texture(TextureHelper.encodeTextureString(textureURL));
+        return this;
+    }
+
+    /**
+     * Set the texture with a pre-encoded string
+     *
+     * @param texture Base64 string of the json of texture location
+     * @throws IllegalArgumentException if the texture string is invalid
+     */
+    public @NotNull SkullBuilder texture(@NotNull String texture) {
+        TextureValidator.validateTextureString(texture);
+        this.texture = texture;
+        return this;
+    }
+
+    /**
+     * Builds the final {@link ItemStack} of {@link Material#PLAYER_HEAD} as long as
+     *
+     * @return The constructed head
+     */
+    public @NotNull ItemStack build() {
+        Preconditions.checkNotNull(texture);
+        Preconditions.checkArgument(name != null || head != null);
+
+        @NotNull SkullMeta meta = (SkullMeta) (new ItemStack(Material.PLAYER_HEAD, 1)).getItemMeta();
+        @NotNull PlayerProfile profile = Bukkit.createProfile(new UUID(0, 0), head);
+
+        profile.setName(TextureHelper.toQualifiedName(category, head == null ? name : head));
+        if (name != null) {
+            meta.setDisplayName(name);
+        } else {
+            meta.setDisplayName(StringUtil.capitaliseSentence(head));
+        }
+
+        profile.setProperty(new ProfileProperty("textures", texture));
+        meta.setPlayerProfile(profile);
+        @NotNull ItemStack head = new ItemStack(Material.PLAYER_HEAD, quantity);
```

Comment:
No need for `@NotNull` annotations in code - unless you want to add them, kotlin can't do anything useful with them since they're not externally visible.

Reply:
Its more for reading and notes, and also because when you auto add them it adds them everywhere